dask/array/routines.py

```diff
@@ -404,6 +405,71 @@ def ediff1d(ary, to_end=None, to_begin=None):
     return r
 
 
+def _gradient_kernel(f, grad_varargs, grad_kwargs):
+    return np.gradient(f, *grad_varargs, **grad_kwargs)
+
+
+@wraps(np.gradient)
+def gradient(f, *varargs, **kwargs):
+    f = asarray(f)
+
+    if not all([isinstance(e, Number) for e in varargs]):
+        raise NotImplementedError("Only numeric scalar spacings supported.")
+
+    if varargs == ():
+        varargs = (1,)
+    if len(varargs) == 1:
+        varargs = f.ndim * varargs
+    if len(varargs) != f.ndim:
+        raise TypeError(
+            "Spacing must either be a scalar or a scalar per dimension."
+        )
+
+    kwargs["edge_order"] = math.ceil(kwargs.get("edge_order", 1))
+    if kwargs["edge_order"] > 2:
+        raise ValueError("edge_order must be less than or equal to 2.")
+
+    drop_result_list = False
+    axis = kwargs.pop("axis", None)
+    if axis is None:
+        axis = tuple(range(f.ndim))
+    elif isinstance(axis, Number):
+        drop_result_list = True
+        axis = (axis,)
+
+    for e in axis:
+        if not isinstance(e, Integral):
+            raise TypeError("%s, invalid value for axis" % repr(e))
+        if not (-f.ndim <= e < f.ndim):
+            raise ValueError("axis, %s, is out of bounds" % repr(e))
+
+    if len(axis) != len(set(axis)):
+        raise ValueError("duplicate axes not allowed")
+
+    axis = tuple(ax % f.ndim for ax in axis)
+
+    if issubclass(f.dtype.type, (np.bool8, Integral)):
+        f = f.astype(float)
+    elif issubclass(f.dtype.type, Real) and f.dtype.itemsize < 4:
+        f = f.astype(float)
+
+    r = [
+        f.map_overlap(
+            _gradient_kernel,
+            dtype=f.dtype,
+            depth={j: 1 if j == ax else 0 for j in range(f.ndim)},
```

Comment:
I think that you might be able to say just `depth={ax: 1}` here

Reply:
Was not sure if that was supported. Though looking at `ghost`, it does seem to be.

Reply:
Went to do this locally and run the tests, but got test failures. Came up with an MRE to reproduce this and filed it as issue ( https://github.com/dask/dask/issues/3444 ). That said, agree this simplification would be nice.